File: 3R/24911_RobotInstructions.cpp
#include <iostream>
#include <string>
#include <sstream>
using namespace std;

int main(){
    int cases, n, sum;
    string str;
    cin >> cases;

    while(cases--){
        cin >> n;
        cin.ignore();
        int arr[n+1];
        
        sum = 0;
        for(int i=1; i<=n; i++){
            getline(cin, str);
            if(str == "LEFT") arr[i] = -1;
            else if(str == "RIGHT") arr[i] = 1;
            else{
                string str2 = str.substr(8);
                istringstream isst(str2);
                int tmp; isst >> tmp;
                arr[i] = arr[tmp];
            }
            sum += arr[i];
        }
        cout << sum << endl;
    }
}

File: 3R/24911_RobotInstructions.java
import java.util.*;

class Main{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
        int cases = sc.nextInt();
        while(cases-->0){
            int n = sc.nextInt();
            sc.nextLine();
            int arr[] = new int[n+1];

            String str = sc.nextLine();
            int sum = 0;
            for(int i=1; i<=n; i++){
                if(str.equals("LEFT")) arr[i] = -1;
                else if(str.equals("RIGHT")) arr[i] = 1;
                else{
                    int tmp = str.substring(8);
                    arr[i] = arr[tmp];
                }
                sum += arr[i];
            }
            System.out.println(sum);
        }
	}
}